### PR TITLE
publish debian packages to jfrog for development purposes

### PR DESCRIPTION
**Pull request by reubenmiller**
Signed-off-by: Reuben Miller <reuben.miller@softwareag.com>

## Proposed changes

Publish the built Debian packages to an private Debian repository (controlled via GitHub secrets). This makes it easier to install a specific non-official release for development and testing purposes.

It also gain experience with the Debian repository to prepare for having a public repository. The documentation on how to use the repository is not included in this PR due to the url being private.

The packages are published in the `build-workflow`, and on any changes to `main` or `release-*` branches

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: .github/workflows/build-workflow.yml [hunk 1]
```diff
@@ -37,20 +37,32 @@ jobs:
           name: debian-packages-amd64
           path: target/debian/*.deb
 
+      - name: Publish debian packages (for internal use)
```
Comment (PradeepKiruvale) [inline .github/workflows/build-workflow.yml:40]: Now the workflow name has to be `build-and-publish.yml`

Comment (reubenmiller) [reply .github/workflows/build-workflow.yml:40]: Yes that is probably worth while to do to make it more transparent. I originally decided against it as it was only publishing internally, however it makes it easier to find if it includes the publishing in the name.

File: .github/workflows/build-workflow.yml [hunk 2]
```diff
@@ -61,28 +73,43 @@ jobs:
...
+          path: target/${{ matrix.job.target }}/debian/*.deb
 
       - name: Upload sawtooth publisher as zip
         # https://github.com/marketplace/actions/upload-a-build-artifact
         uses: actions/upload-artifact@v3
         with:
-          name: sawtooth-publisher_${{ matrix.target }}
-          path: target/${{ matrix.target }}/release/sawtooth-publisher
+          name: sawtooth-publisher_${{ matrix.job.target }}
+          path: target/${{ matrix.job.target }}/release/sawtooth-publisher
```
Comment (didier-wenzek) [inline .github/workflows/build-workflow.yml:93]: We will need to check if this test tool is still used. I think it's useless to maintain it.

File: .github/workflows/build-workflow.yml [hunk 3]
```diff
@@ -61,28 +73,43 @@ jobs:
...
 
       - name: Upload dummy-plugin as zip
         # https://github.com/marketplace/actions/upload-a-build-artifact
         uses: actions/upload-artifact@v3
         with:
-          name: tedge-dummy-plugin_${{ matrix.target }}
```
Comment (didier-wenzek) [inline .github/workflows/build-workflow.yml:87]: We will need to check if this test tool is still used. I think it's useless to maintain it.

**Review by didier-wenzek — APPROVED** [review]
Approved

File: .github/workflows/build-workflow.yml (hunk 2 above)
Comment (reubenmiller) [reply .github/workflows/build-workflow.yml:93]: Yes it is currently used in the `ci_pipeline` workflow. But I have a roadmap item which will look at consolodating the workflows and removing unncessary artifacts.

File: .github/workflows/build-workflow.yml (hunk 1 above)
Comment (reubenmiller) [reply .github/workflows/build-workflow.yml:40]: After reviewing the renaming together we decided to change the job name, not the workflow name.

Work is planned to refine/optimize the build/ci/pr-check/release process where the workflow naming will be revisited.

File: .github/workflows/build-workflow.yml (hunk 3 above)
Comment (reubenmiller) [reply .github/workflows/build-workflow.yml:87]: I will add a ticket to re-evaluate the `tedge-dummy-plugin` to see if it can be removed or not.

**Pull request merged**